main.go

```diff
@@ -1,11 +1,35 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"context"
+	"log"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	tf5server "github.com/hashicorp/terraform-plugin-go/tfprotov5/server"
+	tfmux "github.com/hashicorp/terraform-plugin-mux"
 	"github.com/hashicorp/terraform-provider-tfe/tfe"
 )
 
+const (
+	tfeProviderName = "registry.terraform.io/hashicorp/tfe"
+)
+
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: tfe.Provider})
+	ctx := context.Background()
+	mux, err := tfmux.NewSchemaServerFactory(
+		ctx, tfe.Provider().GRPCProvider, tfe.PluginProviderServer,
```

Comment:
I'd prefer to add a comment to explain than to preface it with `Complex`. "Complex" doesn't tell much to the reader besides "get ready, its complicated" 😂 , which...may be a nice warning!

But I'll put a comment here and where I declare `PluginProviderServer` so that when someone looks there, they can see a full explanation.